src/cookbook/ScottPlot.Cookbook/Locate.cs

```diff
@@ -10,66 +10,96 @@ namespace ScottPlot.Cookbook
-            AppDomain
-            .CurrentDomain
-            .GetAssemblies()
+        private static readonly IRecipe[] _recipes = AppDomain.CurrentDomain.GetAssemblies()
             .SelectMany(s => s.GetTypes())
             .Where(x => x.IsAbstract == false)
             .Where(x => x.IsInterface == false)
             .Where(p => typeof(IRecipe).IsAssignableFrom(p))
             .Select(x => (IRecipe)Activator.CreateInstance(x))
             .ToArray();
 
-        public static IRecipe GetRecipe(string id) => GetRecipes().Where(x => x.ID == id).First();
+        private static readonly string[] _categories = _recipes.Select(r => r.Category).Distinct().ToArray();
+        private static Dictionary<string, IRecipe[]> _recipesByCategory = new Dictionary<string, IRecipe[]>();
+        private static Dictionary<string, IRecipe> _recipesById = new Dictionary<string, IRecipe>();
+        private static List<KeyValuePair<string, IRecipe[]>> _categorizedRecipeList = new List<KeyValuePair<string, IRecipe[]>>();
 
-        public static string[] GetCategories() => GetRecipes().Select(x => x.Category).Distinct().ToArray();
-
-        public static string[] GetCategoriesInDisplayOrder() => GetCategorizedRecipes().Select(x => x.Key).ToArray();
+        private static readonly string[] topCategories =
+        {
+            "Quickstart",
+            "Axis and Ticks",
+            "Advanced Axis Features",
+            "Multi-Axis",
+        };
 
-        public static IRecipe[] GetRecipes(string category) => GetRecipes().Where(x => x.Category == category).ToArray();
+        private static readonly string[] bottomCategories =
+        {
+            "Style",
+            "Misc"
+        };
 
-        public static List<KeyValuePair<string, IRecipe[]>> GetCategorizedRecipes()
+        static Locate() // A static constructor runs exactly once and before the class or an instance of it is needed
         {
-            var categorizedRecipeList = new List<KeyValuePair<string, IRecipe[]>>();
+            InitializeDictionaries();
+            InitializeCategoryList();
+        }
 
-            foreach (string category in GetCategories())
+        private static void InitializeDictionaries()
+        {
+            Dictionary<string, List<IRecipe>> byCategory = new Dictionary<string, List<IRecipe>>();
+            foreach (IRecipe curr in GetRecipes())
             {
-                var recipesForCategory = new KeyValuePair<string, IRecipe[]>(category, GetRecipes(category));
-                categorizedRecipeList.Add(recipesForCategory);
+                _recipesById[curr.ID] = curr;
+
+                if (!byCategory.ContainsKey(curr.Category))
+                {
+                    byCategory.Add(curr.Category, new List<IRecipe>());
+                }
+
+                byCategory[curr.Category].Add(curr);
             }
 
-            string[] topCategories =
+            foreach (string category in GetCategories())
```

Comment:
If we go further, then in fact all we need to do is sort the existing dictionary.
This can be done by defining an additional selector:
```c#
Func<string, int> CategoryIndex(string category)
{
if ( topCategories.Contains(category))
  return 0;
if (bottomCategories.Contains(category))
  return 2;
return 1; // other Category
}
```

add then just sort using this selector
```c#
_recipesByCategory = byCategory.OrderBy(CategoryIndex);
```
I'm not sure if the above code will work (browser programming) but I think you get the idea.
Then it will be possible to throw out(simplify) all the code below `InitializeCategoryList()`

Reply:
That code works, with minor adjustments. Main thing is it needs to take a `KeyValuePair<string, IRecipe>` parameter, not `string`. And then we need to maintain the order within the categories, so that's just a `ThenBy(...)`

Reply:
This _almost_ obviates the need for the static constructor, but since this initialization depends on `_recipesByCategory` this has to run afterwards. There are other solutions such as initializing it the first time it's needed, but for now I kept the static constructor.